app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -87,4 +83,68 @@ public void onDisplayNetworkError(boolean networkNotAvailable) {
     public void onRequestFinish(@NotNull UiFinishReason uiFinishReason) {
         finish();
     }
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the ok/close button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorClose() {
+        onRequestFinish(UiFinishReason.UNABLE_TO_LOAD_SPA);
+    }
+
+    // -----WebUiNetworkErrorFragment.Listener Overrides -----
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the retry button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorRetry() {
+        loadWebUi();
+    }
+
+    private void loadWebUi() {
+        if (this.isFinishing()) {
+            return;
+        }
+        // reuses the fragment if it already existed
+        if (getFragment(TAG_WEB_FRAGMENT) != null) {
+            setFragment((WebUiFragment) getFragment(TAG_WEB_FRAGMENT), TAG_WEB_FRAGMENT);
```

Comment:
Same comment as in the display network error method - when reusing a fragment, we should still set in the args bundle freshly before adding it in the FragmentManager

Reply:
maybe in this case (specifically for the webFragment) we actually don't want to set in the args again - i don't remember the specifics of the implementation in the fragment, but we'd just have to make sure that when the web fragment gets re-added to the container, that it picks up the navigation where the user left off, rather than starting fresh from the dashboard (if possible)

Reply:
updated, but its not reseting the bundle, so whatever state the fragment was at, should still be where it was